### ISSUE TITLE
Invalid proxy_pass with $upstream variable

### ISSUE THREAD

**Issue by shakamus** [jc21/nginx-proxy-manager#1460]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**
After proxy host config update, custom locations fails.
Checking the proxy host number.conf before and after the update i can see this difference:

Before:
```
location ^~ /jenkins/ {
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Scheme $scheme;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Real-IP $remote_addr;
proxy_pass http://127.0.0.01:8080/jenkins/;
}
```
After:

```
location ^~ /jenkins/ {
set $upstream http://127.0.0.01:8080/jenkins/;
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Scheme $scheme;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-For $remote_addr;
proxy_set_header X-Real-IP $remote_addr;
proxy_pass $upstream;
}

```
The same ocurrs with all other custom locations in the proxy host.

**Nginx Proxy Manager Version**
v2.9.9

**To Reproduce**

**Expected behavior**

With the old proxy_pass http://.... works fine, but with variable set $upstream at begining of the location the proxy don't loads all related paths with custom locations

**Screenshots**

With new $upstream variable:
![image](https://user-images.githubusercontent.com/1176404/136047423-f8def03b-1653-422b-aa05-9bc90dff2085.png)

With old config or reventing the .conf manually and reloading nginx:
![image](https://user-images.githubusercontent.com/1176404/136048090-04a38e25-aa61-43f8-9a56-2755422c6da4.png)

**Operating System**
Linux AMI

**Additional context**

Proxy host config files, before and after the last update.

[proxy_conf_new.conf.txt](https://github.com/jc21/nginx-proxy-manager/files/7286820/proxy_conf_new.conf.txt)
[proxy_conf_old.conf.txt](https://github.com/jc21/nginx-proxy-manager/files/7286821/proxy_conf_old.conf.txt)

**Comment by shakamus** [comment]
Some info from stackoverflow that could help.

https://stackoverflow.com/questions/46230270/nginx-using-variable-in-proxy-pass-breaks-routing

**Comment by chaptergy** [comment]
Unfortunately I am not able to find a way which would allow both the usage of a variable for skipping domain caching and a working and generalizable path inclusion.  
As the StackOverflow question also describes, `proxy_pass` expects a _full_ url if it includes a variable, so no path is added to the host. This behaviour is also described in [this article](https://dev.to/danielkun/nginx-everything-about-proxypass-2ona#let-nginx-start-even-when-not-all-upstream-hosts-are-available).  
I have tried just adding the `$request_uri` variable to no avail, I've also tried using the regex location name which did not work and would not generalize well, and I've also tried a [suggestion from a forum](https://www.ruby-forum.com/t/proxy-pass-with-variable-removes-uri/239573/7) to use `rewrite` which did not work either.

Unless some fix is found which would fix both this issue and the one described in the commit, we would have to decide which issue is more relevant, either this one or the caching issue. I would personally say this issue is more relevant and to revert commit 
https://github.com/jc21/nginx-proxy-manager/commit/6c1ae77a2a40283c5444c2f7e0a7c8227aeecbd2. @jc21 

Though the changes added in the commit could possibly also mitigate the issue raised in https://github.com/jc21/nginx-proxy-manager/issues/633.
